### ISSUE TITLE
Why this code needs recovering?

### ISSUE THREAD

**Issue by mariuszs** [RobWin/javaslang-circuitbreaker#2]
```java
import io.github.robwin.failsafe.FailSafe;
import io.github.robwin.retry.Retry;
import javaslang.control.Try;

public class Foobar {

    public static void main(String[] args) {

        Try.run(FailSafe.ofCheckedRunnable(() -> {
            System.out.println("Success");
        }).withRetry(Retry.ofDefaults())
                .decorate())
                .recover((throwable -> {
                    System.out.println("Recover me!");
                    return null;
                }));
        System.out.println("Finish");

    }
}
```

Execution result:

```
Success
Success
Success
Recover me!
Finish
```

**Comment by RobWin** [comment]
Hallo,

 Thx. I have a look at it.

**Comment by RobWin** [comment]
Its a bug :) that happens if you have many negative test cases to test the retry functionality but forgot to write a positive test cases :)

**Comment by RobWin** [comment]
Yes, there is a break missing in 
https://github.com/RobWin/javaslang-circuitbreaker/blob/master/src/main/java/io/github/robwin/retry/Retry.java#L63

Supplier should work.

**Comment by RobWin** [comment]
I also update Javaslang to v2.0 as soon AS it is released

**Comment by mariuszs** [comment]
Thanks, good job!

**Comment by RobWin** [comment]
Should be fixed now. I added tests.

I will release v0.2.0 as soon as Javaslang v2.0 has been released.

**Comment by RobWin** [comment]
Btw I love your catch-exception lib :)

**Comment by RobWin** [comment]
I've released v0.1.7 which fixes this issue.
